internal/webhooks/cluster.go

```diff
@@ -170,6 +172,42 @@ func (webhook *Cluster) validate(ctx context.Context, oldCluster, newCluster *cl
 			),
 		)
 	}
+	if newCluster.Spec.ClusterNetwork != nil {
+		// Ensure that the CIDR blocks defined under ClusterNetwork are valid.
+		if newCluster.Spec.ClusterNetwork.Pods != nil {
+			if err := validateCIDRBlocks(newCluster.Spec.ClusterNetwork.Pods.CIDRBlocks); err != nil {
+				allErrs = append(
+					allErrs,
+					field.Invalid(
+						specPath.Child("clusterNetwork", "services", "cidrBlocks"),
+						spew.Sprint(newCluster.Spec.ClusterNetwork.Pods),
```

Comment:
```suggestion
						newCluster.Spec.ClusterNetwork.Pods.CIDRBlocks,
```

nit: can we maybe just use the value directly or do we need spew? (for the two cases where we simply want to print a string array)
(same for services below)

I would avoid introducing new dependencies to spew which is unmaintained since 5 years if we don't have to. Just to make it easier to get rid of it eventually.

Reply:
Spew was needed in the previous structure because we were nil-checking the chain of fields inside the validation function. Now that this is nil checked before running validation it's a good idea to get rid of it.

Thansk!